lib/assessment.js

```diff
@@ -40,6 +83,24 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // the object passed in as an argument or the object created
 var addMethod = function addMethod(propertyName, method, obj) {
 
+  // define var result as an instance of a new object
+  var result = {};
+
+  var newObj = {};
+
+  var addOrReplaceMethod = function() {
+    var newMethod = method;
+    return this.propertyName[i].newMethod = method;
+  }
+
+    if (/*obj is unset*/) {
+      var newObj = obj;
+      } else {
+      // return the already-in-existence object;
+    }
+
+  return result;
+
```

Comment:
Not sure I can follow your logic on this one. 
Consider the following solution - does it make sense to you?
```javascript
if (!obj) {
  obj = {};
}
obj[propertyName] = method;
return obj;